notebooks/video_clips/VideoClips.ipynb

```diff
@@ -0,0 +1,963 @@
+    "        }\n",
+    "        }]\n",
+    "    fd = open( video_path, 'rb')\n",
+    "    video_data = fd.read()\n",
+    "    fd.close()\n",
+    "\n",
+    "    info_video_query=[{\n",
+    "        \"FindVideo\":{\n",
+    "            \"constraints\": {\n",
+    "                \"id\": [\"==\",str(video_id)]\n",
+    "            },\n",
+    "            \"results\": {\n",
+    "                \"all_properties\":True\n",
+    "            }\n",
+    "        }\n",
+    "    }]\n",
+    "\n",
+    "    run_query(db,add_video_query,[video_data],\"Video Adding\")\n",
+    "    res = run_query(db,info_video_query,None,\"Video Find\")\n",
+    "    video_data= res[0]['FindVideo']['entities'][0]\n",
+    "    width = video_data['_frame_width']\n",
+    "    height = video_data['_frame_height']\n",
+    "    \n",
+    "    add_detections( db, video_id, detections_path, width,height )\n",
+    "    add_clips( db, video_id,  norman_finished.values() )\n",
+    "        "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "94e45efd-1d25-424e-94f4-5fc8fa5f9c87",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    u.remove_all_objects()\n",
+    "    add_video( c, \"norman.mp4\", \"output/norman/detections.csv\" , \"Norman the dog rides a bike with some help\")\n",
+    "    \n",
+    "    u.summary()\n",
+    "except Exception as e:\n",
+    "    print(\"Failed adding: \",e)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f9fe349b-c0bd-4d1d-b43c-ad3dd84fc41d",
+   "metadata": {},
+   "source": [
+    "Ok, our data has been added to the database!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "40efdaea-a2a8-4d36-85dc-c8d74bcead4a",
+   "metadata": {},
+   "source": [
+    "# Clip Verification\n",
+    "It looks like we found a lot of the things we wanted to find, a dog, a bicycle, and the lady who was helping the dorg.\n",
+    "\n",
+    "It's odd that we don't see the dog until frame 136 though, what gives?\n",
+    "\n",
+    "Also, our output shows it first registering the dog at frame 120, but then losing it.\n",
+    "\n",
+    "So we will use aperturedb's `Frames` class to help us debug:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6f1d2a10-557b-4fb2-9f37-a5fa3ca7a79e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from aperturedb.Images import Frames\n",
+    "\n",
+    "FRAME_NUMBER=120\n",
+    "VIDEO_FILE=\"norman.mp4\"\n",
+    "import uuid\n",
+    "\n",
+    "def get_frame_id(video,frame):\n",
+    "    video_uuid = uuid.uuid5( uuid.NAMESPACE_URL, video_url.format( video ))\n",
+    "    frame_uuid = uuid.uuid5( video_uuid, frame_url.format( frame ))\n",
+    "    return str(frame_uuid)\n",
+    "\n",
+    "frms=Frames(c)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bd469e3d-c84b-4ce6-ac53-6e24ff47c2f5",
+   "metadata": {},
+   "source": [
+    "## Issue - dropped dog at 120\n",
```

Comment:
It's these query parts that are very interesting ...do run the code and check in with responses